addons/website_sale/models/website.py

```diff
@@ -162,11 +162,12 @@ def _prepare_sale_order_values(self, partner, pricelist):
         salesperson_id = affiliate_id if self.env['res.users'].sudo().browse(affiliate_id).exists() else request.website.salesperson_id.id
         addr = partner.address_get(['delivery', 'invoice'])
         default_user_id = partner.parent_id.user_id.id or partner.user_id.id
+        team_id = self.salesteam_id.id or partner.parent_id.team_id.id or partner.team_id.id
```

Comment:
no need to define a variable for this, just do:
`'team_id': self.salesteam_id.id or partner.parent_id.team_id.id or partner.team_id.id`